codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -670,6 +686,10 @@ public boolean isPushPromiseSent() {
      */
     private final class DefaultEndpoint<F extends Http2FlowController> implements Endpoint<F> {
         private final boolean server;
+        /**
+         * This is an always increasing sequence number used to hash {@link DefaultStream} instances.
+         */
+        private long lastCreatedStreamIdentity;
```

Comment:
It's known that there can not be more than `Integer.MAX_VALUE` unique streams per connection, will it be enough to use `int` instead of `long` here?

Reply:
ok, let's be safe and keep `long`